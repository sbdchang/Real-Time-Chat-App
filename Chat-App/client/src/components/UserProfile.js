import React from 'react';
import PageNavbar from './PageNavbar';
import 'bootstrap/dist/css/bootstrap.min.css';
import avatar from './avatar.png';
const axios = require('axios');

export default class UserProfile extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			username: "hzw",
			date: "Registration Time"
		};
	}

	componentDidMount() {
		// this.setState({
		// 	username: this.props.changedUsername
		// });
		fetch(`http://localhost:8081/users?username=${this.state.username}`, {
			method: "GET" 
		}).then(response => response.json()).then((response) => {
			this.setState({
				date: response.dateRegistered.substr(0, response.dateRegistered.indexOf('T'))
			});
		})
	}

	async changePassword() {
		let messageOne = document.querySelector("#message-1");
		messageOne.textContent = "";
		await fetch(`http://localhost:8081/users/cpw`, {
			method: "POST" 
		}).then((response) => {
			if (document.getElementById('cpw').value === document.getElementById('npw').value) {
				messageOne.textContent = "Password NOT Changed";
			} else {
				messageOne.textContent = "Password Changed";
			}
		})
	}

	render() {	
		return (
			<div className="UserProfile">
				<PageNavbar active="userprofile" />
				<div className="container bestgenres-container">
			      <div className="jumbotron">
		            <div className="h5">{`Username: ${this.state.username}`}</div>
					<div className="h6">{`Regestration Time: ${this.state.date}`}</div>
					<img src={avatar}/>
			      </div>

			      <div className="jumbotron">
				  <input type='text' placeholder="Current Password" id="cpw" className="cpw-input"/>
				  <input type='text' placeholder="New Password" id="npw" className="npw-input"/>
				  <button id="cp" className="cp-btn" onClick={this.changePassword}>Change Password</button>
				  <div className="results-container" id="results">
			    		<p id = "message-1">  </p>
			      </div>
				  <br></br>
				  <input type='text' placeholder="Current Password" id="dcpw" className="cpw-input"/>
				  <button id="da" className="da-btn">Deactivate Account</button>
				  <div className="results-container" id="results">
			    		<p id = "message-2">  </p>
			      </div>
			      </div>
			    </div>
			</div>
		);
	}
}