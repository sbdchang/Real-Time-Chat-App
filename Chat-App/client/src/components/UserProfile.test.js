import React from 'react';
import ReactDOM from 'react-dom';
import { render, screen } from '@testing-library/react';
import UserProfile from './UserProfile';
import renderer from 'react-test-renderer';

it('UserProfile Snapshot  ', () => {
    const tree = renderer.create(<UserProfile />).toJSON();
    expect(tree).toMatchSnapshot();
  });


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<UserProfile />, div);
});

//add test for text elements

//This function fails TODO
/*
test('render jumbotron', () => {
    const { getByLabelText } = render(<UserProfile />);
    const linkElement = getByText('jumbotron');
    expect(linkElement).toBeInTheDocument();
  });
*/
let div;

beforeEach(() => {
  div = document.createElement("div");
});

afterEach(() => {
    ReactDOM.unmountComponentAtNode(div);
  });


