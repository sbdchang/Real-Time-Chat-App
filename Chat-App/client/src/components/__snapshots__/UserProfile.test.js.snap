// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile Snapshot   1`] = `
<div
  className="UserProfile"
>
  <div
    className="PageNavbar"
  >
    <nav
      className="navbar navbar-expand-lg navbar-light bg-light"
    >
      <span
        className="navbar-brand center"
      >
        CIS557
      </span>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        />
      </div>
    </nav>
  </div>
  <div
    className="container bestgenres-container"
  >
    <div
      className="jumbotron"
    >
      <img
        height="50"
        src="avatar.png"
        width="50"
      />
      <div>
        Username: N/A
      </div>
      <div>
        Status: N/A
      </div>
      <div>
        Regestration Time: N/A
      </div>
    </div>
    <div
      className="jumbotron"
    >
      <input
        className="input"
        id="postStatus"
        placeholder="What's up?"
        type="text"
      />
      <button
        className="btn"
        id="cp"
        onClick={[Function]}
      >
        Post Status
      </button>
      <br />
      <input
        className="filetype"
        id="image_inpt"
        onChange={[Function]}
        type="file"
      />
      <div
        className="results-container"
        id="results"
      >
        <p
          id="message-3"
        >
            
        </p>
      </div>
    </div>
    <div
      className="jumbotron"
    >
      <b
        style={
          Object {
            "fontSize": "25px",
          }
        }
      >
         Account Settings 
      </b>
      <br />
      <br />
      <input
        className="input"
        id="cpw"
        placeholder="Current Password"
        type="text"
      />
      <input
        className="input"
        id="npw"
        placeholder="New Password"
        type="text"
      />
      <button
        className="btn"
        id="cp"
        onClick={[Function]}
      >
        Change Password
      </button>
      <div
        className="results-container"
        id="results"
      >
        <p
          id="message-1"
        >
            
        </p>
      </div>
      <button
        className="btn"
        id="da"
        onClick={[Function]}
      >
        Deactivate Account
      </button>
      <div
        className="results-container"
        id="results"
      >
        <p
          id="message-2"
        >
            
        </p>
      </div>
      <button
        className="btn"
        id="lo"
        onClick={[Function]}
      >
        Log Out
      </button>
    </div>
  </div>
</div>
`;
